### PR TITLE
refactor: add types to reveal the wallet changes needed for Zoe

### PR DESCRIPTION
**Pull request by michaelfig**
These types make it clear what changes need to happen in the wallet.

Please forgive the unreviewed API for the wallet backend.  It will be refactored later, but for now, we're trading technical debt for the features it offers.

### CODE REVIEW

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js [hunk 1]
```diff
@@ -108,18 +144,24 @@ export async function makeWallet({
...
+   * @type {Purse}
+   */
   let zoeInvitePurse;
 
   function getSortedValues(map) {
     const entries = [...map.entries()];
     // Sort for determinism.
     const values = entries
-      .sort(([id1], [id2]) => id1 > id2)
+      .sort(([id1], [id2]) => cmp(id1, id2))
```
Comment (erights) [inline packages/dapp-svelte-wallet/api/src/lib-wallet.js:164]: This is more than a typing change. It is a bug fix. Any other semantically significant changes I should notice?

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js [hunk 2]
```diff
@@ -903,112 +958,116 @@ export async function makeWallet({
...
+          let purse;
+          if (purseOrPetname === undefined) {
+            if (!brandToAutoDepositPurse.has(brand)) {
+              // No automatic purse right now.
+              return depositedPK.promise;
+            }
+            // Plop into the current autodeposit purse.
+            purse = brandToAutoDepositPurse.get(brand);
+          } else if (
+            Array.isArray(purseOrPetname) ||
```
Comment (erights) [inline packages/dapp-svelte-wallet/api/src/lib-wallet.js:1006]: If it is an array here, what does that mean? Is it a path?

**Review by erights — APPROVED** [review]
LGTM. But the only observable changes I noticed are the two I comment on. If there are others, please bring them to my attention before merging. Thanks.

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js (hunk 2 above)
Comment (michaelfig) [reply packages/dapp-svelte-wallet/api/src/lib-wallet.js:1006]: Yes, it is a path.

File: packages/dapp-svelte-wallet/api/src/lib-wallet.js (hunk 1 above)
Comment (michaelfig) [reply packages/dapp-svelte-wallet/api/src/lib-wallet.js:164]: You caught the two.  These were changes necessary to make the types work, so I bundled them with the typings.  Next time I do this, I will make separate steps.

Comment (erights) [reply packages/dapp-svelte-wallet/api/src/lib-wallet.js:164]: That would be fine. Also fine is just calling my attention to them so I don't miss them. Thanks.

**Pull request merged**